### DISCUSSION TITLE
Test on HGB

### DISCUSSION THREAD

**Discussion by shirinmous** [pyg-team/pytorch_geometric#5351] [Q&A]
Hi.
I am working with HGBDataset.
There is no validation set for ACM dataset and the sum of train and test sets is 1966.  The labels of test set are -1.
  How can we test our algorithm using this dataset?

**Comment by rusty1s** [comment]
The test sets of the `HGBDataset` are hidden and one would need to upload the test predictions of your algorithm to the leaderboard, see [here](https://www.biendata.xyz/hgb/).

**Reply by EdisonLeeeee** [reply]
Hi, @rusty1s 
The labels of the test set are provided in `HGBDataset`, see this [PR](https://github.com/pyg-team/pytorch_geometric/pull/5233).

**Reply by shirinmous** [reply]
I saw this [PR](https://github.com/pyg-team/pytorch_geometric/pull/5233)
but I think the code didn't change. Still no label for test set

**Reply by rusty1s** [reply]
@EdisonLeeeee Still, the labels are randomly replaced, right? See https://www.biendata.xyz/hgb/#/about.

**Reply by EdisonLeeeee** [reply]
@rusty1s My bad. It seems that I made a wrong PR. Really sorry :(.

**Reply by shirinmous** [reply]
I downloaded ACM from HGB website. There are three  files for labels
label.dat with 907 labels
label.dat.test with 1059 labels
label.dat.test_full with 2118 labels
So there are labels for all 3025 nodes.
I am confused. They provide labels but when we read the dataset by HGBDataset, we can't access the labels.

**Reply by shirinmous** [reply]
In HGBDataset code labels are loaded from label.dat.test but not saved in hetero graph.

I think the code has changed after that [PR](https://github.com/pyg-team/pytorch_geometric/pull/5233l) ,
'for y in test_ys:
                n_id, n_type = mapping_dict[int(y[0])], n_types[int(y[2])]
                if data[n_type].y.dim() > 1:  # multi-label
                    for v in y[3].split(','):
                        data[n_type].y[n_id, int(v)] = 1
                else:
                    data[n_type].y[n_id] = int(y[3])
                data[n_type].test_mask[n_id] = True'
 but when I install torch_geometric the old code is instlled. I don't know why?
`for y in test_ys:
       n_id, n_type = mapping_dict[int(y[0])], n_types[int(y[2])]
      data[n_type].test_mask[n_id] = True
`

**Reply by rusty1s** [reply]
You might need to install PyG from master to have access to the PR.
@EdisonLeeeee Shall we revert the PR to avoid confusion?

**Reply by EdisonLeeeee** [reply]
@rusty1s Sure.
